Gems/EMotionFX/Code/EMotionFX/Source/ActorManager.cpp

```diff
@@ -146,60 +149,82 @@ namespace EMotionFX
     Actor* ActorManager::FindActorByName(const char* actorName) const
     {
         // get the number of actors and iterate through them
-        const auto found = AZStd::find_if(m_actors.begin(), m_actors.end(), [actorName](const AZStd::shared_ptr<Actor>& a)
+        const auto found = AZStd::find_if(
+            m_actorAssets.begin(), m_actorAssets.end(),
+            [actorName](const ActorAssetData& a)
         {
-            return a->GetNameString() == actorName;
+                return a.GetAs<Integration::ActorAsset>()->GetActor()->GetNameString() == actorName;
```

Comment:
You shouldn't need these `GetAs<ActorAsset>()` calls. The type is already `ActorAssetData`, which is a `AZ::Data::Asset<Integration::ActorAsset>`. So I'd expect this to be:
```suggestion
                return a->GetActor()->GetNameString() == actorName;
```